docs(storage): add intro page (#455)
status-im/ift-docs: docs/storage/introduction-to-logos-storage.md

---
slug: /storage

> sidebar_position: 0

Reviewer: **The frontmatter is missing every metadata field the rest of the docs carry.** This page has only `slug` and `sidebar_position`.

The closest analogue in the repo is `docs/lez/introduction-to-the-logos-execution-zone.md` — also a product intro page, also `sidebar_position: 0`, also a bare-product slug (`/lez`) — and it carries the full block:

```
title: Introduction to the Logos Execution Zone
doc_type: concept
product: blockchain
topics: lez
authors: kashepavadan
owner: logos
doc_version: 1
```

Note `doc_type: concept` is also what exempts that page from the `:::tip[Version]` callout that `#457` just added to all 39 `doc_type: procedure` pages — so declaring it here keeps this page correctly out of scope for that convention too, rather than looking like an oversight.

Adding `title` is behaviour-preserving: with no `title`, Docusaurus already derives it from the `# Logos Storage` H1, so the sidebar label and page title stay exactly as they are now.

```suggestion
title: Logos Storage
doc_type: concept
product: storage
topics: storage
authors: gmega
owner: logos
doc_version: 1
slug: /storage
sidebar_position: 0
```

---
_Generated by [Claude Code](https://claude.ai/code)_

---

import Figure from '@site/src/components/Figure';

# Logos Storage

> Logos storage is a filesharing protocol that allows users of the Logos stack to publish and share files in a decentralized manner. It provides persistence guarantees that are similar to those of BitTorrent[^1]; i.e., those of an [organically replicated network](#organic-replication-and-persistence-guarantees), while adding [stronger privacy guarantees](#privacy). Logos storage [is provided as a Logos module](./get-started/run-logos-storage-node.md), and requires [the Logos runtime and tooling](/get-started/what-is-logos) to work.

Reviewer: **`decentralized` → `decentralised`** — house style. Across the merged docs it's **44 "decentralised" to 1 "decentralized"**, and `Logos.BritishSpelling` is `level: error`.

Vale won't flag this one, which is worth knowing: the swap list in `.github/styles/Logos/BritishSpelling.yml` has `decentralize` and `decentralization` but **not the `-ized` past tense**, so the rule has a hole exactly where this page falls. If you want to close it for everyone, adding these two lines near line 26 of that file would do it:

```
  decentralized: decentralised
  decentralizing: decentralising
```

```suggestion
Logos storage is a filesharing protocol that allows users of the Logos stack to publish and share files in a decentralised manner. It provides persistence guarantees that are similar to those of BitTorrent[^1]; i.e., those of an [organically replicated network](#organic-replication-and-persistence-guarantees), while adding [stronger privacy guarantees](#privacy). Logos storage [is provided as a Logos module](./get-started/run-logos-storage-node.md), and requires [the Logos runtime and tooling](/get-started/what-is-logos) to work.
```

---
_Generated by [Claude Code](https://claude.ai/code)_

## What can you do with Logos storage?

> The key functionality of Logos storage is _file sharing_: you can make files avaiable on the network which other people can then download. This is, in some ways, similar to how Google Drive[^3] works (Figure 1): you start by **(1)** [uploading a file to your own node](./get-started/run-logos-storage-node.md#publish-a-file) which will **(2)** store the file internally and **(3)** produce a file **Content IDentifier** (CID), which uniquely identifies your file and works very much like a "share link".

Reviewer: **`avaiable` → `available`** — missing the second `l`. Confirmed it renders verbatim in the built page ("you can make files avaiable on the network").

```suggestion
The key functionality of Logos storage is _file sharing_: you can make files available on the network which other people can then download. This is, in some ways, similar to how Google Drive[^3] works (Figure 1): you start by **(1)** [uploading a file to your own node](./get-started/run-logos-storage-node.md#publish-a-file) which will **(2)** store the file internally and **(3)** produce a file **Content IDentifier** (CID), which uniquely identifies your file and works very much like a "share link".
```

---
_Generated by [Claude Code](https://claude.ai/code)_

> You can then **(4)** share this CID with other users using a third-party mechanism such as e-mail or [Logos messaging](/messaging). Once in posession of a CID, a user $q` can then its Logos storage node to download and store the file **(5)** from the network and store it internally **(6)**. Once the file is available [^2], the user can then retrieve it from the node **(7)** and consume its contents.

Reviewer: ⛔ **This sentence is broken three ways, and all three reach the published page.** I pulled this out of the built HTML rather than reading the source, to confirm it isn't a rendering artefact:

> Once in `posession` of a CID, a user `$q` ` can then its Logos storage node to download and store the file **(5)** …

1. **`$q\`` — a stray `$` and an unmatched backtick.** Because the backtick has no partner, Markdown gives up and prints both characters literally: readers see `$q\`` on the page. This is left over from `cd0a3d8` ("remove math (no support installed)") — the `$…$` was stripped but its contents weren't converted to the `` `q` `` inline-code style the rest of the page uses for `p`, `q`, `F` and `g`.
2. **A verb is missing** — "a user `q` can then *its* Logos storage node to download" has no predicate. Given the numbered walkthrough, "can then **use**" is the intent.
3. **`posession` → `possession`** (double `s`).

I also swapped "its" → "their", since `q` is a person here and the neighbouring paragraph (line 22) already says "*their* local node".

```suggestion
You can then **(4)** share this CID with other users using a third-party mechanism such as e-mail or [Logos messaging](/messaging). Once in possession of a CID, a user `q` can then use their Logos storage node to download and store the file **(5)** from the network and store it internally **(6)**. Once the file is available [^2], the user can then retrieve it from the node **(7)** and consume its contents.
```

---
_Generated by [Claude Code](https://claude.ai/code)_

<Figure id="fig:1" src={require("./assets/logos-intro.png").default} caption="Logos storage workflow." number={1} />

## Organic replication and persistence guarantees

The reader may have noticed from the previous discussion that a user `p` wishing to share a file `F` must first upload it to their local node. A user `q` wishing to obtain `F` must then download `F` to _their_ local node first. The reason is that, underneath, `p` and `q` are engaging in a replication protocol - when `p` initially shares `F`, `F` is added to the network with 1 replica (`p`'s node). Once `q` downloads it `F` now has 2 replicas. If a third user `g` were to download `F` as well, then it would have 3 replicas, and so on.

> Since the number of replicas is determined by the number of nodes _interested_ in `F` and interest is inherently an organic property, we name this _organic replication_. This is in stark contrast to Decentralized Storage Networks such as Swarm[^4] or SIA[^5] (or products like Google Drive) where providers put storage for sale and will take up potentially arbitrary content so long as it is paid for. This also places a fundamental limitation on persistence: if no one is interested in your files, chances are that losing your node means your data is lost as well.

Reviewer: Same British-spelling point as line 10 — **`Decentralized Storage Networks` → `Decentralised`**.

```suggestion
Since the number of replicas is determined by the number of nodes _interested_ in `F` and interest is inherently an organic property, we name this _organic replication_. This is in stark contrast to Decentralised Storage Networks such as Swarm[^4] or SIA[^5] (or products like Google Drive) where providers put storage for sale and will take up potentially arbitrary content so long as it is paid for. This also places a fundamental limitation on persistence: if no one is interested in your files, chances are that losing your node means your data is lost as well.
```

---
_Generated by [Claude Code](https://claude.ai/code)_

Note that this is not an insurmountable limitation: one could set up multiple nodes and create replication groups for files using Logos messaging, but this is currently not provided out-of-the-box. For files of public interest, however, replication should happen naturally: parties interested in keeping some content online can operate their own nodes and curate the list of files they wish to replicate. This allows the operator to control the type of content they host, as well as how much content they host.

## Privacy

> A key problem with filesharing networks such as BitTorrent - and Logos storage itself - is that organic replication exposes both the party that provide content (providers) and the party that downloads it (downloaders) to censorship. Even with encrypted communications, active attackers can join the network and uncover which nodes are doing what.

Reviewer: **"the party that provide content" → "provides"** — subject/verb agreement. "party" is singular, and the parallel clause in the same sentence already gets it right ("the party that *downloads* it").

While here: the two ` - ` dashes in this sentence are hyphens doing an em-dash's job. The repo is inconsistent about this so I won't push it, but since the line is being touched anyway, `—` reads better and matches the em-dashes used on lines 22 and 32.

```suggestion
A key problem with filesharing networks such as BitTorrent — and Logos storage itself — is that organic replication exposes both the party that provides content (providers) and the party that downloads it (downloaders) to censorship. Even with encrypted communications, active attackers can join the network and uncover which nodes are doing what.
```

---
_Generated by [Claude Code](https://claude.ai/code)_

> Privacy in Logos storage has been focused on _unlinking_ both providers and downloaders by the use of Mix networks[^6]. This is strongly inspired by how Tribler[^7] works, but adapted to Logos' infrastructure and constraints. As of the [Logos testnet v0.2](https://roadmap.logos.co/testnets/v02), we are able to provide some level of unlinkability to content lookup queries and, by [testnet v0.3](https://roadmap.logos.co/testnets/v03), we should be able to provide unlinkable downloads; i.e., it should be possible for a downloader `q` to download a file `F` without revealing to neither passive nor active attackers that they are, in fact, downloading `F`.

Reviewer: **"without revealing to neither passive nor active attackers" is a double negative**, which strictly reverses the meaning you want — as written it says the download *is* revealed to them.

`without` already carries the negation, so the correlative pair needs to be `either … or`:

```suggestion
Privacy in Logos storage has been focused on _unlinking_ both providers and downloaders by the use of Mix networks[^6]. This is strongly inspired by how Tribler[^7] works, but adapted to Logos' infrastructure and constraints. As of the [Logos testnet v0.2](https://roadmap.logos.co/testnets/v02), we are able to provide some level of unlinkability to content lookup queries and, by [testnet v0.3](https://roadmap.logos.co/testnets/v03), we should be able to provide unlinkable downloads; i.e., it should be possible for a downloader `q` to download a file `F` without revealing to either passive or active attackers that they are, in fact, downloading `F`.
```

---
_Generated by [Claude Code](https://claude.ai/code)_

> The ultimate goal is _fully anonynous filesharing_; i.e., a system in which neither providers nor downloaders can be linked to the content they are providing/downloading.

Reviewer: **`anonynous` → `anonymous`** — `n` where the `m` belongs. This one is especially worth catching because it lands on the page's closing statement of intent, in italics.

```suggestion
The ultimate goal is _fully anonymous filesharing_; i.e., a system in which neither providers nor downloaders can be linked to the content they are providing/downloading.
```

---
_Generated by [Claude Code](https://claude.ai/code)_
